### PR TITLE
Type hinting for functions

### PR DESCRIPTION
**Pull request by SHAKOTN**
# #139 <--- 

## **Done:**
- Functions that are covered now with typehints:
```
to_merkle_format
bcvx_claims_snapshot
bcvxcrv_claims_snapshot
distribute_rewards_to_snapshot
get_sett_multipliers
get_user_multipliers
boost_sett
calc_sushi_distribution
convert_to_merkle_tree
propose_root
manage_root
fetch_all_schedules
generate_rewards_in_range
fetch_block_by_timestamp
token_diff_table
combine_rewards
get_last_proposed_cycle
calc_next_cycle_range
```
- Lots of PEP8 improvements in Python
- Sorting out imports

### **What is not done:**
Functions like:
```
fetch_tree
fetch_current_tree
fetch_pending_tree
fetch_current_merkle_data
fetch_pending_merkle_data
```
are pretty sensitive and we don't have enough test coverage on those, making changing types of those quite scary. 
_**What I suggest:**_ come back to typehinting those functions(or even wrapping their output into dataclasses) when we have better test coverage and it's not so scary to change those types

### CODE REVIEW

File: rewards/calc_rewards.py [hunk 1]
```diff
@@ -87,17 +95,19 @@ def propose_root(
...
     save=False,
-):
+) -> None:
     """
     Propose a root on a chain
 
     :param chain: chain to propose root
     :param start: start block for rewards
     :param end: end block for rewards
     :param save: flag to save rewards file locally, defaults to False
```
Comment (mitche50) [inline rewards/calc_rewards.py:108]: save duplicated here

Comment (SHAKOTN) [reply rewards/calc_rewards.py:108]: Not really, one entry is describing param, enother one is describing the type of that param

File: rewards/classes/RewardsList.py [hunk 2]
```diff
@@ -89,24 +94,23 @@ def getTokenRewards(self, user, token):
...
-                nodeEntry["tokens"].append(tokenAddress)
-                nodeEntry["cumulativeAmounts"].append(str(int(cumulativeAmount)))
-                intAmounts.append(int(cumulativeAmount))
+        int_amounts = []
+        for tokenAddress, cumulativeAmount in user_data.items():
+            node_entry["tokens"].append(tokenAddress)
+            node_entry["cumulativeAmounts"].append(str(int(cumulativeAmount)))
+            int_amounts.append(int(cumulativeAmount))
 
         # console.print(
```
Comment (mitche50) [inline rewards/classes/RewardsList.py:115]: TODO: Remove unused commented code

File: rewards/classes/RewardsList.py [hunk 3]
```diff
@@ -89,24 +94,23 @@ def getTokenRewards(self, user, token):
         else:
             return 0
 
-    def to_node_entry(self, user, userData, cycle, index):
+    def to_node_entry(self, user, user_data, cycle, index) -> Tuple[Dict[str, Any], str]:
         """
```
Comment (mitche50) [inline rewards/classes/RewardsList.py:98]: TODO: Add type hinting and param definitions to comments

File: rewards/explorer.py [hunk 4]
```diff
@@ -11,7 +14,7 @@
 }
 
 
-def fetch_block_by_timestamp(chain: str, timestamp: int):
+def fetch_block_by_timestamp(chain: str, timestamp: int) -> Optional[Dict]:
```
Comment (mitche50) [inline rewards/explorer.py:17]: API / HTTP requests should be handled through a function which catches any non-200 responses and retries with maximum amount of retries.  If retries are exhausted, log to discord and gracefully exit.

**Review by mitche50 — APPROVED** [review]

**Pull request merged**